docs/DeveloperGuide.adoc

```diff
@@ -218,30 +235,360 @@ image::CommitActivityDiagram.png[]
+If it exists, a command exception is thrown saying that there is already a group
+with the given group code.
+
+The interactions between the logic and model components when adding a group are similar
+to the interactions when deleting a group as shown below.
+
+The following sequence diagram shows the interactions between the logic and model
+components when the user inputs the command 'group delete m/CS2103 g/G03'.
+
+Note: This diagram is under the case where a group with the group code G03 does exist
+in the module with module code CS2103 inside the TA-Tracker.
+
+.Group Delete - Sequence Diagram
+image::DeleteGroupSequenceDiagram.png[]
+
+1. LogicManager uses the TATrackerParser to first parse the user command.
+
+2. The TATrackerParser sees that the command is of type group and passes the
+command to the GroupCommandParser.
+
+3. The GroupCommandParser sees that the command is of type delete and passes the
+arguments to the DeleteGroupCommandParser.
+
+4. The DeleteGroupCommandParser creates a Module with the given module code and
+a group with the given group code.
+
+5. The DeleteGroupCommandParser then creates a DeleteGroupCommand object and passes
+it the created module and group. The parser then returns the DeleteGroupCommand
+
+6. LogicManager calls DeleteGroupCommand's execute method. The DeleteGroupCommand object
+checks whether a module with the given module code already exists in TA-Tracker.
+If it doesn't, a command exception is thrown saying that a module with the given module
+code doesn't exist in the TA-Tracker.
+
+7. If the module exists, the DeleteGroupCommand object retrieves the module from the
+model and checks whether the module has a group with the given group code. If it doesn't,
+a command exception is thrown saying that no such group exists. If the group does
+exist, it is removed from the module.
+
+8. The SortGroupCommand returns a CommandResult with a success message.
+
+==== Implementation of the Sort Command
+
+The sort command allows the user to sort the students in the module view either alphabetically
+or by rating.
+
+The sort command can be used in three ways:
+
+1. sort g/GROUP_CODE m/MODULE_CODE t/TYPE : When a user enters the command in this
+manner, they are sorting all the students of the given group in the given module
+by type TYPE (which can be either alphabetical or by rating).
+
+2. sort g/MODULE_CODE t/TYPE : When a user enters a command in this manner, they
+are sorting all the students of all the groups in the given module by type TYPE
+(which can be either alphabetical or by rating).
+
+3. sort t/TYPE : When a user enters a command in this manner, they are sorting all
+students of all groups of all the modules in the TA-Tracker by the type TYPE (which
+can be either alphabetical or by rating).
+
+Since these sort commands function differently but use the same parser, the following
+class structure is used.
+
+.Sort Commands - Class Diagram
+image::SortCommandsClassDiagram.png[]
```

Comment:
The words here are a little small to see (compared to the other diagrams) so maybe can change the sizing of the words!